iris-mpc-cpu/src/hawkers/aby3/aby3_store.rs

```diff
@@ -285,19 +285,24 @@ impl VectorStore for Aby3Store {
         self.lift_distances(dist).await.unwrap()[0].clone()
     }
 
-    #[instrument(level = "trace", target = "searcher::network", skip_all, fields(batch_size = vectors.len()))]
+    #[instrument(level = "trace", target = "searcher::network", skip_all, fields(queries = queries.len(), batch_size = vectors.len()))]
     async fn eval_distance_batch(
         &mut self,
-        query: &Self::QueryRef,
+        queries: &[Self::QueryRef],
         vectors: &[Self::VectorRef],
     ) -> Vec<Self::DistanceRef> {
         if vectors.is_empty() {
             return vec![];
         }
         let vectors = self.storage.iter_vectors(vectors).await;
-        let pairs = vectors
+        let pairs = queries
             .iter()
-            .map(|vector| (&query.processed_query, &**vector))
+            .flat_map(|q| {
+                vectors
+                    .iter()
+                    .map(|vector| (&q.processed_query, &**vector))
+                    .collect::<Vec<_>>()
+            })
```

Comment:
This flattens it here, but does not unflatten into per query results. Is this fine to leave up to the caller or is the internal unflattening a nicer interface?

Reply:
At the moment, `eval_distance_batch` is usually followed by `less_than_batch` or `is_match_batch` that accept flattened vectors. And the current design in `hawk_main` does unflattening.

Reply:
This does look incorrect or at least surprising to me too. This does a flatten / Cartesian product, since it is only called with one element in this PR.

I think this should be documented and have a different name than just "_batch".

Reply:
Then we can use the function in follow up PRs.

Reply:
Ok that's what the PR description said :)

Reply:
What action do you want me to do here? For clarity, I can unflatten the result right here. And in a later PR, we can change it if necessary.

Reply:
Nothing, I just misunderstood your intention before.